util/flightcontrol/flightcontrol.go

```diff
@@ -27,12 +27,29 @@ type Group struct {
 	m  map[string]*call // lazily initialized
 }
 
+type retries struct {
+	num     int
+	backoff time.Duration
+}
+
 // Do executes a context function syncronized by the key
 func (g *Group) Do(ctx context.Context, key string, fn func(ctx context.Context) (interface{}, error)) (v interface{}, err error) {
+	retries := retries{backoff: time.Second}
+	v, err = g.do(ctx, key, fn, &retries)
+	// if retries.num > 0 {
+	// 	fmt.Println(retries.num)
+	// }
```

Comment:
left over debug?

Also, please could the commit message acquire an explanation of why this is a desirable change, what the current failure mode is etc.

Reply:
Yes, lets not keep it.